freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UmsState.java

```diff
@@ -77,14 +81,24 @@ public UsersState getUsersState(String environmentCrn) {
                 builder.addMemberToGroup(adminGrpName, user.getName());
             });
 
-        // TODO filter machine users by environment rights
+        machineUserToGroupsMap.entrySet()
+            .forEach(e -> {
+                FmsUser user = umsMachineUserToUser(e.getKey());
+                e.getValue().forEach(g -> {
+                    FmsGroup group = umsGroupToGroup(g);
+                    crnToGroup.put(g.getCrn(), group);
+                    builder.addMemberToGroup(group.getName(), user.getName());
+                    builder.addGroup(group);
+                });
+
+            });
         machineUserMap.entrySet()
                 .forEach(e -> {
                     FmsUser fmsUser = umsMachineUserToUser(e.getValue());
                     builder.addUser(fmsUser);
```

Comment:
can put this up in the machineUserToGroupsMap.entrySet().forEach above. If we have Users and MachineUsers as the keys to the map then we don't need the extra machineUserMap.